### ISSUE TITLE
Documents revision with OneFlow 0.5.0

### ISSUE THREAD

**Issue by doombeaker** [Oneflow-Inc/oneflow-documentation#316]
文档网站计划改版，主要两点变化：
1. 原有内容以 lazy 模式为主；新版本将以 eager 为主
2. 原有内容主要针对有其它框架经验的用户，没有覆盖对深度学习框架不熟悉的用户；新版本将增加针对没有深度学习框架经验用户的内容

以下是供讨论和未来分工的大纲

----------------------------------------------------------
#### 2.2.2 如何得到一个 Tensor

- 从 numpy 构造
- 使用 Tensor 中的方法( `norm_` 等)
- 使用算子（`zeros` 等）

#### 2.2.4 常见的 Tensor 操作（算子）

- 数值计算类的
- 位置操作类
- 兼有的

#### 	2.5.1 `backward()` 自动求导

  - `requires_grad` 与 `backward()`
  - 为什么只能是针对标量求导
  - 为什么默认保留叶子节点的导数

  #### 2.5.2 `grad` 成员及其相关操作

  - detach
  - retain

  #### 2.5.3 利用自动求导手工实现反向传播

  #### 2.5.4 利用 `flow.optim` 中已有的类进行反向传播

**Comment by MARD1NO** [comment]
2.2.2 如何得到一个 Tensor 中
```
使用 Tensor 中的方法( norm_ 等) -> 这部分不是很能理解？是想表达在网络里的数据都是Tensor吗，如果是这样的话感觉这部分可以简洁，说明tensor是oneflow的基本数据类型即可
```

2.3 OFRecord 数据集的加载与使用
```
这部分直接扔进cookbook里吧？我理解我们后面eager的dataloader应该是另外一套，不需要制作为OFRecord格式，这点迟哥应该需要和露阳确认下
```

**Comment by BBuf** [comment]
2.1.3 反向传播 要不改成模型训练吧，这样包含反向传播和参数更新

**Comment by Flowingsun007** [comment]
- 感觉2.1的2.1.1 【下载训练数据】没啥存在的必要，快速上手我们一定是提供开箱即用包含数据下载/处理流程的傻瓜式代码的
- 可以单独搞一个和【2.3 OFRecord 数据集的加载与使用】并列的小结，类似于：【数据处理和dataloader】介绍我们eager下对齐pytorch的dataloader/dataset的用法、常见的数据处理方法。

**Comment by Flowingsun007** [comment]
- 3. 分布式训练 开头感觉需要加一个类似于快速上手的demo，让别人可以比较轻松的跑一下试试

**Comment by doombeaker** [comment]
>  使用 Tensor 中的方法( norm_ 等) -> 这部分不是很能理解？是想表达在网络里的数据都是Tensor吗，如果是这样的话感觉这部分可以简洁，说明tensor是oneflow的基本数据类型即可
是除了从 numpy 构造得到tensor对象以外，还有一些其它方法可以得到 tensor 对象，入 ones， zeros 等，这些常见操作需要介绍的。
----
> 这部分直接扔进cookbook里吧？我理解我们后面eager的dataloader应该是另外一套，不需要制作为OFRecord格式，这点迟哥应该需要和露阳确认下
> 这部分直接扔进cookbook里吧？我理解我们后面eager的dataloader应该是另外一套，不需要制作为OFRecord格式，这点迟哥应该需要和露阳确认下

以上两点合并在一起，有道理，我把目录改了，具体的规划到时留给露阳做吧（可能要等露阳写完 dataset 才可以出文档）

----
> 感觉2.1的2.1.1 【下载训练数据】没啥存在的必要，快速上手我们一定是提供开箱即用包含数据下载/处理流程的傻瓜式代码的

这个是参考 PyTorch 的 https://pytorch.org/tutorials/beginner/basics/quickstart_tutorial.html：

![image](https://user-images.githubusercontent.com/3351623/125196853-fc887700-e28d-11eb-85cb-ac6c83a296a6.png)


当然写的时候不会强调是“下载数据集/模型”，这里规划是指调用的接口和对应的内容，其实上做的是这个事情

----

> 分布式训练 开头感觉需要加一个类似于快速上手的demo，让别人可以比较轻松的跑一下试试

很有道理，我加一个3.0，分布式训练 quick start

**Comment by doombeaker** [comment]
要做的事情：
1. 根据以上大纲，重写 docs.oneflow.org 内容
2. 为文档配上恰当的动图或讲解视频（如使用 https://github.com/3b1b/manim 制作）
3. 为文档仓库加入自动化测试（基于 doctest，需要试水)

涉及的技能要求：
1. 了解 mkdocs 文档生成工具
2. 熟悉 markdown
3. 熟悉 Python
4. 熟悉 PyTorch 或者 OneFlow，了解常见的算子、API
5. 对深度学习训练中的一些进阶话题，如分布式训练，超大规模网络中的困难有了解
6. 有写技术笔记的习惯，乐于整理技术文档

现在倾向于找实习生配合我完成以上的主要工作，实习生可能来自目前已经在岗的，也可能来自社区。

工作方式和时间：需要集中的工作1~2个月。工作流程包括：
1. 我来指定大纲，强调文档要突出的重点，和实习生讨论文章、示例代码的写作要点
2. 实习生执行，我及其它同事review，迭代修改，直到产出文章
3. 中间遇到的写作、技术问题，由我负责组织培训讲解，包括但是不限于：相关编程工具、库的使用；与 OneFlow 有关的需要介绍的特性；其它深度学习框架的技术特点

这项工作强调自愿与 **投入**，需要参与人像对待技术项目一样严谨和认真（事实上产出的过程也接近中低难度的技术项目）。
有意者请联系我：
OneFlow 的实习生：slack 上联系
社区同学：qq 群里的 later，或者 yaochi@oneflow.org

**Comment by yuanms2** [comment]
可以按照这个要求，招聘一下实习生

**Comment by doombeaker** [comment]
已经合并 https://github.com/Oneflow-Inc/oneflow-documentation/pull/347
